aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+mod storage_wrapper;
+pub(crate) mod vm_wrapper;
+
+use crate::{
+    adapter_common::{preprocess_transaction, PreprocessedTransaction},
+    block_executor::vm_wrapper::AptosVMWrapper,
+    AptosVM,
+};
+use aptos_aggregator::{delta_change_set::DeltaOp, transaction::TransactionOutputExt};
+use aptos_block_executor::{
+    errors::Error,
+    executor::{BlockExecutor, RAYON_EXEC_POOL},
+    output_delta_resolver::{OutputDeltaResolver, ResolvedData},
+    task::{
+        Transaction as BlockExecutorTransaction,
+        TransactionOutput as BlockExecutorTransactionOutput,
+    },
+};
+use aptos_logger::debug;
+use aptos_state_view::StateView;
+use aptos_types::{
+    state_store::state_key::StateKey,
+    transaction::{Transaction, TransactionOutput, TransactionStatus},
+    write_set::{WriteOp, WriteSet, WriteSetMut},
+};
+use move_core_types::vm_status::VMStatus;
+use rayon::prelude::*;
+use std::collections::HashMap;
+
+impl BlockExecutorTransaction for PreprocessedTransaction {
+    type Key = StateKey;
+    type Value = WriteOp;
+}
+
+// Wrapper to avoid orphan rule
+pub(crate) struct AptosTransactionOutput(TransactionOutputExt);
```

Comment:
Not a big fan of avoiding the orphan rule which is there for a reason (and not a technical one). It makes perhaps notations a bit more convenient, but generally the code harder to understand. Not saying this needs to be differently done, but just an opinion.

Reply:
I really dislike this as well and tried to get rid of it, can solve some issues but the problem I couldn't easily overcome is a potential circular dependency between aptos-aggregator (where TransactionOutputExt is defined) and block-executor crates (that defines traits, and also uses aggregators).. Hopefully we will get to a place where we restructure this wrapper, or in general restructure the crates and factor out the common types (like we do currently at the aptos-core level, something similar maybe we could do the same at the aptos-vm or aptos-move).